### ISSUE TITLE
VS Code intellisence seems to ignore diagnostic suppressors

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/vscode-csharp#5845]
## Environment data
`dotnet --info` output:
<details>

```
ПАКЕТ SDK ДЛЯ .NET:
 Version:   8.0.100-preview.5.23303.2
 Commit:    3fe444af72

Среда выполнения:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.100-preview.5.23303.2\

.NET workloads installed:
Нет установленных рабочих нагрузок для отображения.

Host:
  Version:      8.0.0-preview.5.23280.8
  Architecture: x64
  Commit:       bc78804f5d

.NET SDKs installed:
  6.0.410 [C:\Program Files\dotnet\sdk]
  7.0.107 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]
  7.0.400-preview.23274.1 [C:\Program Files\dotnet\sdk]
  8.0.100-preview.5.23303.2 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.0-preview.5.23302.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.0-preview.5.23280.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.0-preview.5.23302.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```
</details>

VS Code version: 1.79.2
C# Extension version: v2.0.212 (+ C# Dev Kit)

## OmniSharp log
I don't know where to see it. I have 4 output channels related to C#. Here is content of channel named just `C#` (again, not sure this is omnisharp log):
<details>

```
Dotnet path: c:\Users\<username>\AppData\Roaming\Code\User\globalStorage\ms-dotnettools.vscode-dotnet-runtime\.dotnet\7.0.7\dotnet.exe
Activating C# + C# Dev Kit + C# IntelliCode...
info: LanguageServerHost[0]
      Starting server...

```
</details>

## Steps to reproduce

1. Create a simple C# project on .NET 7
2. Add EF Core v7 to the project
3. Add some basic `User` class
4. Create db context with the following content:
```cs
using Microsoft.EntityFrameworkCore;

namespace TestEFApp
{
    public class ApplicationDbContext : DbContext
    {
        public DbSet<User> Users { get; set; }
    }
}
```

## Expected  behavior
No warnings, because EF Core v7 by default comes with analyzers package, which contains diagnostic suppressor for nullability of `DbSet` properties

## Actual behavior
I see warning:
![Code_xqrpfoPkdP](https://github.com/dotnet/vscode-csharp/assets/70431552/7c71fa5c-a711-49d1-8db2-577d7897e379)

## Additional context
1. In big VS everything works as expected and no warning is shown:
![devenv_k7PXLxbmwm](https://github.com/dotnet/vscode-csharp/assets/70431552/b8944ea9-5431-4754-9d72-8c759641c15a)
2. Here is full sample project: [TestEFApp.zip](https://github.com/dotnet/vscode-csharp/files/11820992/TestEFApp.zip)

**Comment by dibarbet** [comment]
@mavasani is this something you could help take a look at?

**Comment by mavasani** [comment]
@DoctorKrolic I am  unable to reproduce this on my end. I tried both the released and pre-release version of https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csdevkit and also tried our daily internal drop of the extension. I don't see the mentioned nullable warning in live analysis or during build command.

![image](https://github.com/dotnet/vscode-csharp/assets/10605811/9b838f2e-21b9-4774-8cb1-3fb2af0eafbe)

I would have thought this should be fixed with [#68270](https://github.com/dotnet/roslyn/pull/68270), earlier we used to intermittently skip running suppressors in VSCode (LSP pull diagnostics).
